#  ДЗ #1 

## задача №1
На складе лежат разные фрукты в разном количестве.
Нужно написать функцию, которая на вход принимает любое количество названий фруктов и их количество, а возвращает общее количество фруктов на складе

In [52]:
def hmf(num):
    print(num)
    kor = {}
    fr_sum = 0
    for i in range(num):
        name = input('fruit name: ')
        number = int(input(f"input number of {name}: "))
        kor.update({name: number})
        fr_sum += number
    print(kor)
    print(f"total fruit number is: {fr_sum}")
    
    # решение через словарь
    sum2 = 0
    for k in kor.values():
        sum2 += k
    print('total fruit number is:', sum2)

N = int(input("how much diff fruits?:"))
hmf(N)

KeyboardInterrupt: Interrupted by user

## Задача №2
Дан список с затратами на рекламу. Но в данных есть ошибки, некоторые затраты имеют отрицательную величину. Удалите такие значения из списка и посчитайте суммарные затраты
[100, 125, -90, 345, 655, -1, 0, 200]
Используйте list comprehensions

In [58]:
cost = [100, 125, -90, 345, 655, -1, 0, 200]
new_cost = [i for i in cost if i >= 0]
new_cost

[100, 125, 345, 655, 0, 200]

[100, 125, 345, 655, 0, 200]

## Задача №3
Даны два списка.
Дата покупки
['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']

Суммы покупок по датам
[1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

### Задача №3.1
Найдите, какая выручка у компании в ноябре
Используйте list comprehensions

In [65]:
purch_date = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', 
              '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', 
              '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', 
              '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
purch_sum =  [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 
              2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]
len(purch_date), len(purch_sum)


(30, 30)

In [71]:
import time

fmt = "%Y-%m-%d"

def nov2(purch_date, purch_sum):
    nov_sum = 0
    count = 0
    month = 11
    for d in purch_date:
        dt = time.strptime(d, fmt)
        if dt[1] == month: 
            # print('ind, nov_sum + purch= ', purch_date.index(d), nov_sum, "+", purch_sum[count], "=") # выдает корявые индексы
            print('ind, nov_sum + purch= ', count, nov_sum, "+", purch_sum[count], "=")
            nov_sum += purch_sum[count]            
            print(nov_sum)
        count += 1
    print(f'выручка за {month}-ый месяц = {nov_sum}') 

nov2(purch_date, purch_sum)



ind, nov_sum + purch=  7 0 + 4944 =
4944
ind, nov_sum + purch=  9 4944 + 3701 =
8645
ind, nov_sum + purch=  12 8645 + 7037 =
15682
ind, nov_sum + purch=  13 15682 + 4274 =
19956
ind, nov_sum + purch=  19 19956 + 2004 =
21960
ind, nov_sum + purch=  20 21960 + 2822 =
24782
ind, nov_sum + purch=  26 24782 + 316 =
25098
выручка за 11-ый месяц = 25098


ИТОГ

ind, nov_sum + purch=  7 0 + 4944 =
4944
ind, nov_sum + purch=  9 4944 + 3701 =
8645
ind, nov_sum + purch=  12 8645 + 7037 =
15682
ind, nov_sum + purch=  13 15682 + 4274 =
19956
ind, nov_sum + purch=  19 19956 + 2004 =
21960
ind, nov_sum + purch=  20 21960 + 2822 =
24782
ind, nov_sum + purch=  26 24782 + 316 =
25098
выручка за 11-ый месяц = 25098

### Задача №3.2
Найдите выручку компании в зависимости от месяца
Для этого напишите функцию, которая на вход принимает список с датами и список с выручкой, а на выходе словарь, где ключи - это месяцы, а значения - это выручка.
Используйте аннотирование типов.

In [73]:
def mon_revenue3(purch_date: list, purch_sum: list ) -> list:
    t_m = {}
    count = 0
    n_num = 0
    num = 0
    ind = 0
    for d in purch_date:
        dt = time.strptime(d, fmt)
        mon = dt[1]
        num = purch_sum[ind]
        if t_m.get(mon) == None:
            n_num = 0
        else:
            n_num = t_m.get(mon)
        n_num = num + n_num
        t_m.update({mon : n_num})
        # print(ind, ' t_m =', t_m)
        ind += 1

    print('Выручка по месяцам = ', t_m)
mon_revenue3(purch_date, purch_sum)

Выручка по месяцам =  {9: 25647, 12: 45452, 10: 28645, 11: 25098}


Выручка по месяцам =  {9: 25647, 12: 45452, 10: 28645, 11: 25098}